SULLesson10_12/SULLesson10_12/FiltersScreen/FiltersScreen.swift

```diff
@@ -0,0 +1,193 @@
+//  FiltersScreen.swift
+//  SULLesson10_12
+//
+//  Created by Григоренко Александр Игоревич on 29.01.2023.
+//
+
+import SwiftUI
+
+/// Экран с настройкой фильтрации результатов поиска
+struct FiltersScreen: View {
+
+    // MARK: - Private constants
+
+    private enum Constants {
+        static let cateforyTextViewText = "Category"
+        static let moreViewText = "More"
+        static let moreViewImageName = "chevron.right"
+        static let pricingSliderViewText = "Pricing"
+        static let colorViewText = "Colors"
+        static let maxPriceText = "100$"
+        static let minPriceText = "5000$"
+    }
+
+    // MARK: - Public properties
+
+    var body: some View {
+        VStack {
+            contentView
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(LinearGradient(colors: [Color.yellow, Color.red], startPoint: .leading, endPoint: .trailing))
+    }
+
+    // MARK: - Private properties
+
+    private let columns = [GridItem(), GridItem(), GridItem(), GridItem()]
+
+    @StateObject private var viewModel = FilterScreenViewModel()
+    @State private var pricingValue = 0.0
+
+
+    private var contentView: some View {
+        VStack(spacing: 0) {
+            headTitleView
+                .background(Color.white)
+            categoryScrollView
+                .background(Color.white)
+            pricingSliderView
+                .background(Color.white)
+            colorsView
+                .background(Color.white)
+        }
+        .padding(.top)
+    }
+
+    private var headTitleView: some View {
+        HStack {
+            categoryTextView
+            Spacer()
+            moreView
+        }
+    }
+
+    private var categoryTextView: some View {
+        Text(Constants.cateforyTextViewText)
+            .font(.title)
+            .bold()
+            .padding(.leading)
+    }
+
+    private var moreView: some View {
+        HStack {
+            Text(Constants.moreViewText)
+                .foregroundColor(.gray.opacity(0.4))
+                .font(.title)
+                .bold()
+            Image(systemName: Constants.moreViewImageName)
+                .resizable()
+                .modifier(showMoreImageModifier())
+        }
+        .padding([.top, .bottom])
+    }
+
+    private var categoryScrollView: some View {
+        ScrollView(.horizontal, showsIndicators: false) {
+            categoryViewsInStackView
+        }
+        .scaledToFit()
+        .offset(y: -25)
+    }
+
+    private var categoryViewsInStackView: some View {
+        HStack(spacing: 0) {
+            ForEach(self.viewModel.imageNames, id: \.self) { imageNames in
+                categoryView(imageName: imageNames)
+            }
+        }
+        .padding(.top)
+    }
+
+    private func categoryView(imageName: String) -> some View {
```

Comment:
```suggestion
    private func makeCategoryView(imageName: String) -> some View {
```
функции ниже прочерти